src/main.rs

```diff
@@ -1,48 +1,32 @@
-// Copyright 2019 Twitter, Inc.
+// Copyright 2019-2020 Twitter, Inc.
 // Licensed under the Apache License, Version 2.0
 // http://www.apache.org/licenses/LICENSE-2.0
 
-#![deny(clippy::all)]
-#![allow(clippy::new_ret_no_self)]
-
 #[macro_use]
 extern crate logger;
 
-mod common;
-mod config;
-mod samplers;
-mod stats;
+#[macro_use]
+extern crate failure;
 
-use crate::common::*;
-use crate::config::Config;
-use crate::samplers::*;
+use std::sync::Arc;
 
-use atomics::{AtomicBool, AtomicPrimitive, Ordering};
 use logger::Logger;
-use metrics::{Metrics, Reading};
-use slab::Slab;
-use timer::Wheel;
+use metrics::*;
+use tokio::runtime::Builder;
 
-use std::sync::{Arc, Mutex};
-use std::thread;
-use std::time::{Duration, Instant};
-
-struct Stats {
-    sequential_timeouts: usize,
-}
+mod common;
+mod config;
+mod exposition;
+mod samplers;
 
-impl Default for Stats {
-    fn default() -> Self {
-        Self {
-            sequential_timeouts: 0,
-        }
-    }
-}
+use common::*;
+use config::Config;
+use samplers::*;
 
-#[allow(clippy::cognitive_complexity)]
-fn main() {
+fn main() -> Result<(), Box<dyn std::error::Error>> {
     // get config
     let config = Config::new();
+    let config = Arc::new(config);
```

Comment:
Do you need to shadow here?

Reply:
I will fix that. I believe arc is necessary, but can be a single definition.

Reply:
Fixed